sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala

```diff
@@ -117,7 +117,7 @@ private[sql] object DataFrame {
 @Experimental
 class DataFrame private[sql](
     @transient val sqlContext: SQLContext,
-    @DeveloperApi @transient val queryExecution: SQLContext#QueryExecution)
+    @DeveloperApi @transient var queryExecution: SQLContext#QueryExecution)
```

Comment:
What I'm proposing is something like this:

```scala
class DataFrame private[sql](
    @transient val sqlContext: SQLContext,
    @DeveloperApi @transient private var _queryExecution: SQLContext#QueryExecution)
  extends RDDApi[Row] with Serializable {
  ...
  @DeveloperApi def queryExecution: SQLContext#QueryExecution = _queryExecution
  ...
}
```

In this way, we don't change the API (by adding a setter introduced by the public `var`).